plugins/woocommerce-admin/client/utils/xstate/find-component.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * Does a depth-first search of a meta object to find the first instance of a component.
+ *
+ * @template T - The type of the component meta object
+ */
+export function findComponentMeta< T >(
```

Comment:
The code was kind of cryptic so I asked ChatGPT to review it. Possible issues:

1. `for .. in` loop without `hasOwnProperty` check
2. Possible infinite loop by passing self-referencing objects due to recursive call

It [suggested code](https://chat.openai.com/share/345a1081-2c29-4abc-8059-a4d965b3db8f), but I did not test this:

```ts
export function findComponentMeta<T>(
    obj: Record<string, unknown>,
    visited = new Set<Record<string, unknown>>()
): T | undefined {
    if (visited.has(obj)) {
        return undefined;
    }

    visited.add(obj);

    for (const key in obj) {
        if (obj.hasOwnProperty(key)) {
            if (key === 'component') {
                // Additional checks can be added here to ensure that the value under 'component' meets your specific criteria
                return obj as T;
            } else if (typeof obj[key] === 'object' && obj[key] !== null) {
                const found = findComponentMeta<T>(
                    obj[key] as Record<string, unknown>,
                    visited
                );
                if (found !== undefined) {
                    return found;
                }
            }
        }
    }

    return undefined;
}
```

Reply:
we’ve gone one full circle, chatgpt wrote it and reviewed it 😂

Reply:
Thanks! I've verified the improvements as well as .. added more test cases with chatgpt ...